backend/drm/util.c

```diff
@@ -323,10 +324,14 @@ static bool match_obj_(struct match_state *st, size_t skips, size_t score, size_
 	return match_obj_(st, skips, score, replaced, i + 1);
 }
 
-size_t match_obj(size_t num_objs, const uint32_t objs[static restrict num_objs],
-		size_t num_res, const uint32_t res[static restrict num_res],
-		uint32_t out[static restrict num_res]) {
-	uint32_t solution[num_res];
+size_t match_obj(size_t num_objs, const uint32_t *objs,
+		size_t num_res, const uint32_t *res,
+		uint32_t *out) {
+	uint32_t *solution = malloc(sizeof(uint32_t) * num_res);
+	if (solution == NULL) {
+		wlr_log(WLR_ERROR, "Failed to allocate memory.");
+		return 0;
```

Comment:
Should we make this ssize_t and return -ENOMEM to propagate? There's only 2 call sites that don't use the return value at all.

Reply:
Not sure what the practice is here so I'll wait for somebody's input.

Reply:
`ssize_t` is only guaranteed to be able to hold `size_t` plus -1, nothing more. Maybe return -1 to indicate failure?